helix-term/src/commands.rs

```diff
@@ -584,6 +585,34 @@ fn goto_first_nonwhitespace(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
+fn trim_selections(cx: &mut Context) {
+    let (view, doc) = current!(cx.editor);
+    let text = doc.text().slice(..);
+
+    let selection = doc.selection(view.id).clone().transform(|range| {
+        let mut start = range.from();
+        let mut end = range.to();
+
+        start = movement::skip_while(text, start, |x| x.is_whitespace()).unwrap_or(start);
+        if end > 0 {
+            end = movement::backwards_skip_while(text, end - 1, |x| x.is_whitespace())
+                .unwrap_or(end)
+                + 1;
+        }
+
+        if start > end {
```

Comment:
This happens only when the whole range is whitespace right? Those ranges should be removed altogether, you can use ranges = filter_map+collect instead of transform to construct a new selection. (You'll need to check !ranges.is_empty?() and fail because a selection needs at least one range)